### PR TITLE
Add single peer connection support

### PR DESCRIPTION
**Pull request by starwarfan**

### CODE REVIEW

File: scripts/patches/licode/0019-Add-Rtp-demuxer.patch [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
++  std::map<std::string, uint32_t> mapping_ssrcs_;
+ };
+ 
+ }  // namespace erizo
+diff --git a/erizo/src/erizo/rtp/RtpExtensionProcessor.cpp b/erizo/src/erizo/rtp/RtpExtensionProcessor.cpp
+index 6ea37e8..6e1ae73 100644
+--- a/erizo/src/erizo/rtp/RtpExtensionProcessor.cpp
++++ b/erizo/src/erizo/rtp/RtpExtensionProcessor.cpp
+@@ -19,6 +19,7 @@ RtpExtensionProcessor::RtpExtensionProcessor(const std::vector<erizo::ExtMap> ex
+   translationMap_["urn:3gpp:video-orientation"] = VIDEO_ORIENTATION;
```
Comment (taste1981) [inline scripts/patches/licode/0019-Add-Rtp-demuxer.patch:104]: why extension id is fixed here?

Comment (starwarfan) [reply scripts/patches/licode/0019-Add-Rtp-demuxer.patch:104]: ID is read from SDP

File: source/agent/webrtc/wrtcConnection.js [hunk 2]
```diff
@@ -138,56 +232,65 @@ class WrtcStream {
...
+  var wrtcId = spec.connectionId;
+  var wrtc = null;
+  var threadPool = spec.threadPool;
+  var ioThreadPool = spec.ioThreadPool;
+  var networkInterfaces = spec.network_interfaces;
+
+  var remoteSdp = null;
+  var localSdp = null;
+  // mid => { operationId, type, sdpDirection, formatPreference, rids, enabled, finalFormat }
+  var operationMap = new Map();
```
Comment (jianjunz) [inline source/agent/webrtc/wrtcConnection.js:246]: An operation is a transceiver?

**Review by jianjunz — COMMENTED** [review]
I'm not familiar with the server side code, only briefly took a look at the WebRTC part.

Please also run tests on Firefox and Safari if QA has bandwidth to do so.

File: source/agent/webrtc/wrtcConnection.js (hunk 2 above)
Comment (starwarfan) [reply source/agent/webrtc/wrtcConnection.js:246]: is a publication or subscription, multiple transceivers

File: source/agent/webrtc/wrtcConnection.js [hunk 3]
```diff
@@ -41,13 +32,102 @@ const TransportSeqNumUri =
...
-    this.audioFramePacketizer = config.audioFramePacketizer;
-    this.videoFrameConstructor = config.videoFrameConstructor;
-    this.videoFramePacketizer = config.videoFramePacketizer;
+class WrtcStream extends EventEmitter {
+
+  /*
+   * audio: { format, ssrc, mid, midExtId }
+   * video: { format, ssrc, mid, midExtId, transportcc, red, ulpfec }
+   */
+  constructor(id, wrtc, direction, {audio, video}) {
```
[the remark this answers is not in the document]
Comment (starwarfan) [reply source/agent/webrtc/wrtcConnection.js:41]: It's a send stream or receive stream, wrtc is the 'connection' object defined in 'connection.js'

**Review by taste1981 — APPROVED** [review]
Merge this first for basic testing.

**Comment by jianjunz** [comment]
This PR added some duplicated code. For example, `portal/client.js` is similar to `portal/v11Client.js`, `portal/versions/requestDataValidator.js` is similar to `portal/requestDataValidator.js`. It's better to reuse existing code, or refactor code instead of adding duplicate code.

**Pull request merged**